week0003/mattfanz/php/src/FinTech/Action.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/*
+ * This file is part of the FinTech.
+ */
+
+namespace FinTech;
+
+/**
+ * Class Action
+ * @package FinTech
+ */
+class Action
+{
+    /**
+     * @var string
+     */
+    protected $action;
+
+    /**
+     * @var string
+     */
+    protected $name;
+
+    /**
+     * @var double
+     */
+    protected $amount;
+
+    /**
+     * @var AccountManager
+     */
+    protected $accountManager;
+
+    /**
+     * @var string
+     */
+    protected $output;
+
+    const ACTION_ADD_CLIENT = 'A';
+    const ACTION_DEPOSIT_MONEY = 'D';
+    const ACTION_WITHDRAW_MONEY = 'W';
+
+    /**
+     * Action constructor.
+     * @param $action
+     * @param $name
+     * @param $amount
+     * @param AccountManager $accountManager
+     */
+    public function __construct($action, $name, $amount, AccountManager $accountManager)
+    {
+        $this->action = $action;
+        $this->name = $name;
+        $this->amount = $amount;
+        $this->accountManager = $accountManager;
+        
+        switch ($this->action) {
```

Comment:
This must be not in the constructor! What if you decide not to perform the action?

Move it to the performAction function.